packages/geoprocessing/src/toolbox/geoblaze/rasterStats.ts

```diff
@@ -75,28 +87,37 @@ export const rasterStats = async (
   let defaultStats: StatsObject[] = [];
   for (let i = 0; i < numBands; i++) {
     defaultStats[i] = {};
-    for (let j = 0; j < stats.length; j++) {
-      if (stats[j] === "area") defaultStats[i][stats[j]] = 0;
-      else defaultStats[i][stats[j]] = defaultStatValues[stats[j]];
+    // Initialize default values for published stats
+    for (let j = 0; j < statsToPublish.length; j++) {
+      if (statsToPublish[j] === "area") {
+        defaultStats[i][statsToPublish[j]] = 0;
+      } else if (
+        statsToPublish[j] === "histogram" &&
+        categorical &&
+        categoryMetricValues
+      ) {
+        let hist = {};
+        categoryMetricValues.forEach((c) => (hist[c] = 0)); // load zero for each histogram category
+        defaultStats[i][statsToPublish[j]] = hist;
+      } else {
+        defaultStats[i][statsToPublish[j]] =
+          defaultStatValues[statsToPublish[j]];
+      }
```

Comment:
If there's categoryMetricValues defined, I enhanced the histogram default value, by adding a value of 0 for every category